pkg/autoops/domain/auto_ops_rule.go

```diff
@@ -95,47 +140,108 @@ func (a *AutoOpsRule) SetOpsType(opsType proto.OpsType) {
 	a.AutoOpsRule.TriggeredAt = 0
 }
 
+func (a *AutoOpsRule) SetAutoOpsStatus(status proto.AutoOpsStatus) {
+	a.AutoOpsRule.AutoOpsStatus = status
+	a.AutoOpsRule.UpdatedAt = time.Now().Unix()
+}
+
 func (a *AutoOpsRule) AddOpsEventRateClause(oerc *proto.OpsEventRateClause) (*proto.Clause, error) {
 	ac, err := ptypes.MarshalAny(oerc)
 	if err != nil {
 		return nil, err
 	}
-	return a.addClause(ac)
+	clause, err := a.addClause(ac, oerc.ActionType)
+	if err != nil {
+		return nil, err
+	}
+	a.AutoOpsRule.UpdatedAt = time.Now().Unix()
+	a.AutoOpsRule.TriggeredAt = 0
+	return clause, nil
 }
 
 func (a *AutoOpsRule) AddDatetimeClause(dc *proto.DatetimeClause) (*proto.Clause, error) {
 	ac, err := ptypes.MarshalAny(dc)
 	if err != nil {
 		return nil, err
 	}
-	return a.addClause(ac)
+	clause, err := a.addClause(ac, dc.ActionType)
+	a.sortDatetimeClause()
+
+	if err != nil {
+		return nil, err
+	}
+	a.AutoOpsRule.UpdatedAt = time.Now().Unix()
+	a.AutoOpsRule.TriggeredAt = 0
```

Comment:
Just confirming. Why do we need to reset the `TriggeredAt`?
Currently, the user can't add any clause after creating it. 
If this function will be used for the multi-schedule, I think we don't need to reset it because the `TriggeredAt` will be removed after the migration.